pennylane/dla/lie_closure.py

```diff
@@ -253,6 +388,13 @@ def __eq__(self, other):
 
         rank1 = np.linalg.matrix_rank(self._M)
         rank2 = np.linalg.matrix_rank(other._M)
-        rank3 = np.linalg.matrix_rank(np.concatenate([self._M, other._M], axis=1))
+
+        # To accommodate the case where the _pw_to_idx have
+        # different permutations, re-arrange ``other`` with the order of ``self``
+        other_M = np.zeros((other._num_pw, other._rank), dtype=float)
+        for i, ps in enumerate(other.basis):
+            for pw, value in ps.items():
+                other_M[self._pw_to_idx[pw], i] = value
+        rank3 = np.linalg.matrix_rank(np.concatenate([self._M, other_M], axis=1))
```

Comment:
Good catch! :tada: 
Probably this is not too performant, I'm afraid? :sweat_smile: 
But we can always come back to try and improve that if it should be necessary..

Reply:
The alternative would be to permute the matrix, but constructing the permutation matrix also requires going through all rows and columns so I think it wouldnt be much faster that way 🤔

Reply:
Mhm, true!
I thought of getting all the permutation indices and then having a single "slicing" call to `other_M`, but that might have been a bit shortsighted :upside_down_face: LGTM!